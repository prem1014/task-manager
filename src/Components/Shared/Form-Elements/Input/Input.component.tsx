import React from 'react';

interface IInputProps {
    type: string;
    changed(event: any): any;
    placeholder?: string;
    value: string;
    className?: string
}

const Input: React.FC<IInputProps> = (props) => {
    return (
        <input 
            type={props.type}
            onChange={props.changed}
            placeholder={props.placeholder}
            value={props.value}
            className={props.className ? props.className : 'form-control'}
        />
    )
}

export default Input;