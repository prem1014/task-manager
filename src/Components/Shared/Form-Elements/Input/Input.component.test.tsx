import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Input from './Input.component';

describe('Input component test suite', () => {
    Enzyme.configure({ adapter: new Adapter() });
    let componentWrapper: any;
    const inputProps = {
        type: 'text',
        changed: jest.fn(),
        value: 'name'
    }

    beforeEach( () => {
        componentWrapper = shallow(<Input {...inputProps}/>);
    });

    it('Input component snapshot test', () => {
        expect(componentWrapper).toMatchSnapshot();
    });
});