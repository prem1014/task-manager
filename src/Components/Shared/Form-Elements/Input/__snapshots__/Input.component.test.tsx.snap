// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component test suite Input component snapshot test 1`] = `ShallowWrapper {}`;
